mk/docs.mk

```diff
@@ -126,6 +126,7 @@ doc/not_found.html: $(D)/not_found.md $(HTML_DEPS) | doc/
 		--markdown-css https://doc.rust-lang.org/rust.css $<
 
 define DEF_DOC
+RUSTDOC_DEPS_$(1) ?=
```

Comment:
I'm a little wary of sprinkling these around as they're often just intended to be empty. I'm not sure we gain a huge amount of benefit from ensuring that 100% of our variables are defined before use (e.g. did this uncover a bug?), so we may want to hold off on these for now.

Changes like `ifeq` to `ifdef` are of course fine though.